src/controllers/qaController.js

```diff
@@ -1,21 +1,74 @@
 const db = require('../db');
 
+const loopAnswersToGetPhotos = (answers, object) => {
+  return Promise.all(answers.map(answer=>{
+    return db.query('SELECT * FROM photos WHERE answer_id = $1',[answer.id])
+      .then(photos=>{
+        answer.photos = photos.rows;
+        return answer;
+      })
+  }));
+};
+
+// this thing needs optimization!!!
+const loopQuestionsToGetAnswers = (questions) => {
+  return Promise.all(questions.map(question=>{
+    return db.query('SELECT * from answers where question_id = $1',[question.id])
+      .then(returnedAnswers => {
+        let returnedQuestion = question;
+        return Promise.all(returnedAnswers.rows.map(answer=>{
+          return db.query('select url,id from photos where answer_id = $1', [answer.id])
+            .then(photos=>{
+              answer.photos = photos.rows;
+              return answer;
+            })
+        }))
+          .then(values=>{
+            const obj = {}
+            for (var answer of values) {
+              obj[answer.id] = answer;
+            }
+            return obj;
+          })
+          .then(finalResult=>{
+            returnedQuestion.answers = finalResult;
+            // console.log(returnedQuestion)
+            return returnedQuestion;
+          })
+          .catch(err=>{throw err})
+        // console.log(returnedAnswers)
+        // let answers = returnedAnswers.rows[0].json_object_agg
+        // let keys = Object.keys(answers);
+        // Promise.all(Object.keys(answers).map(key=>{
+        //   return db.query('select url,id from photos where answer_id = $1', [key])
+        //     .then(photos=>{
+        //       console.log(photos)
+        //       answers[key].photos = photos.rows;
+        //       return answers[key];
+        //     })
+        // }))
+        //   .then(values=>console.log(values))
+      })
+  }));
+};
+
+
+
 exports.getQuestions = (req,res) => {
   const responseObj = {};
   const {product_id, count} = req.query;
 
-  db.query(`SELECT * from questions WHERE product_id = $1 and reported = $2 limit $3`, [product_id, 0, count || 5])
+  db.query('SELECT * FROM questions WHERE product_id = $1 ORDER BY question_date DESC limit $2', [product_id,  count || 5])
```

Comment:
For Q&A, the only sorting that needs to happen is that the more helpful q/a appear first